### DISCUSSION TITLE
Troubleshooting packaging of a WASM application on GitHub Pages

### DISCUSSION THREAD

**Discussion by zdpcdt** [AvaloniaUI/Avalonia#12085] [Q&A]
I've been trying to package our WASM application, Semi.Avalonia, and deploy it on GitHub Pages. I followed the instructions in this file [link to the GitHub Actions workflow file](https://github.com/AvaloniaUI/XamlPlayground/blob/main/.github/workflows/pages.yml), but unfortunately, I encountered an error. The browser console displayed the following error message:

![image](https://github.com/AvaloniaUI/Avalonia/assets/54255897/f8c6fd90-606b-4a0e-8edf-eacf8660c10f)

I'm not sure why this is happening. To explore the generated GitHub Pages site, you can visit the following address: https://zdpcdt.github.io/Semi.Avalonia/

I forked the code from Semi.Avalonia, which can be found at this address: 
https://github.com/zdpcdt/Semi.Avalonia.

 I attempted to package and run the application from the `fonts` branch.

**Comment by hez2010** [comment]
You are not referencing ColorPicker and DataGrid axaml files in your Semi.Avalonia.Demo.Web App.axaml.

Try adding these lines to your [App.axaml](https://github.com/zdpcdt/Semi.Avalonia/blob/main/demo/Semi.Avalonia.Demo.Web/App.axaml):
```xaml
        <StyleInclude Source="avares://Semi.Avalonia.DataGrid/Index.axaml" />
        <StyleInclude Source="avares://Semi.Avalonia.ColorPicker/Index.axaml" />
```

**Reply by rabbitism** [reply]
We did that, the branch we are using is `font`

**Comment by maxkatz6** [comment]
Just in case, is there any StyleInclude involved from the C# code?

**Reply by zdpcdt** [reply]
No, we just involved XAML codes.

**Comment by zdpcdt** [comment]
After some investigation, I discovered that deploying the project in `debug` mode resolved the issue.
It appears that when using the release mode, there might be excessive trimming of assemblies, which could have led to the problem.

You can access the deployed page at the following link:
https://irihitech.github.io/Semi.Avalonia/
